neuraxle/steps/util.py

```diff
@@ -260,6 +258,30 @@ def set_hyperparams_space(self, hyperparams_space: HyperparameterSpace) -> 'Base
     def get_hyperparams_space(self) -> HyperparameterSpace:
         return self.step.get_hyperparams_space()
 
+    def reverse(self) -> 'BaseStep':
+        """
+        The object will mutate itself such that the ``.transform`` method (and of all its underlying objects
+        if applicable) be replaced by the ``.inverse_transform`` method.
+
+        Note: the reverse may fail if there is a pending mutate that was set earlier with ``.will_mutate_to``.
+
+        :return: a copy of self, reversed. Each contained object will also have been reversed if self is a pipeline.
+        """
+        for step in self.steps:
+            step.mutate(new_method="inverse_transform", method_to_assign_to="transform")
```

Comment:
I don't think this is necessary. Shouldn't the reverse in BaseStep already mutate the object properly? I think it already works without reimplementing this method and that the unit test will still work.